exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/Marshaler.java

```diff
@@ -6,15 +6,25 @@
 package io.opentelemetry.exporter.otlp.internal;
 
 import java.io.IOException;
+import java.io.OutputStream;
 
 /**
  * Marshaler from an SDK structure to protobuf wire format.
  *
  * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
  * at any time.
  */
-public interface Marshaler {
-  void writeTo(Serializer output) throws IOException;
+public abstract class Marshaler {
 
-  int getProtoSerializedSize();
+  /** Marshals into the {@link OutputStream} in proto binary format. */
+  public final void writeBinaryTo(OutputStream output) throws IOException {
+    CodedOutputStream cos = CodedOutputStream.newInstance(output);
+    writeTo(new ProtoSerializer(cos));
+    cos.flush();
+  }
+
+  /** Returns the number of bytes this Marshaler will write in proto binary format. */
+  public abstract int getBinarySerializedSize();
+
+  abstract void writeTo(Serializer output) throws IOException;
```

Comment:
We were just talking about interface vs abstract class. Switched to abstract class here to be able to make this package private.